apps/server/src/modules/h5p-editor/service/contentStorage.service.ts

```diff
@@ -0,0 +1,271 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import {
+	ContentId,
+	IContentMetadata,
+	IContentStorage,
+	IFileStats,
+	ILibraryName,
+	IUser,
+	LibraryName,
+	Permission,
+} from '@lumieducation/h5p-server';
+import { Inject, Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { FileDto } from '@src/modules/files-storage/dto';
+import { Readable } from 'stream';
+import { S3ClientAdapter } from '../../files-storage/client/s3-client.adapter';
+import { H5PContent } from '../entity';
+import { H5PContentRepo } from '../repo';
+
+@Injectable()
+export class ContentStorage implements IContentStorage {
+	constructor(
+		private readonly repo: H5PContentRepo,
+		@Inject('S3ClientAdapter_Content') private readonly storageClient: S3ClientAdapter
+	) {}
+
+	public async addContent(
+		metadata: IContentMetadata,
+		content: unknown,
+		user: IUser,
+		contentId?: ContentId | undefined
+	): Promise<ContentId> {
+		try {
+			let h5pContent: H5PContent;
+
+			if (contentId) {
+				h5pContent = await this.repo.findById(contentId);
+				h5pContent.metadata = metadata;
+				h5pContent.content = content;
+			} else {
+				h5pContent = new H5PContent({ metadata, content });
+			}
+
+			await this.repo.save(h5pContent);
+
+			return h5pContent.id;
+		} catch (error) {
+			throw new InternalServerErrorException(error, 'ContentStorage:addContent');
+		}
+	}
+
+	public async addFile(contentId: string, filename: string, stream: Readable, user?: IUser): Promise<void> {
+		this.checkFilename(filename);
+
+		if (!(await this.contentExists(contentId))) {
+			throw new NotFoundException('The content does not exist');
+		}
+
+		const fullPath = this.getFilePath(contentId, filename);
+		const file: FileDto = {
+			name: filename,
+			data: stream,
+			mimeType: 'application/json',
+		};
+
+		await this.storageClient.create(fullPath, file);
+	}
+
+	public async contentExists(contentId: string): Promise<boolean> {
+		try {
+			await this.repo.findById(contentId);
+			return true;
+		} catch (error) {
+			return false;
+		}
```

Comment:
Bitte das catch entfernen, evtl. durch verschieben in das Repo, oder durch eine Repo methode ersetzten ohne OrFail.